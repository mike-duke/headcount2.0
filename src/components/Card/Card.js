import React from 'react';
import './Card.scss';
import PropTypes from 'prop-types';

const Card = ({location, stats, addDistrictToCompare}) => {
  const statsKeys = Object.keys(stats);

  const listItems = statsKeys.map((key) => {
    let color = 'yellow'

    if (stats[key] > 0.5) {
      color = 'purple';
    } 

    return ( 
      <li key={key}>{`${key}: `}
        <span className={color}>
          {`${stats[key]}`}
        </span>
      </li>
    )
  })

  return ( 
    <article className="card" onClick={() => {addDistrictToCompare(location)}}>
      <h3>{location}</h3>
      <hr />
      <ul className="timeframes">
        {listItems}
      </ul>
    </article>
  )
}

Card.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.object.isRequired,
  addDistrictToCompare: PropTypes.func.isRequired
}

export default Card;