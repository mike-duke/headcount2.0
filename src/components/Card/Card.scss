.card {
  border: 2px solid black;
  width: 200px;
  height: 350px;
  background: gray;
  color: white;
  margin: 10px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
  border-radius: 20px;
}

h3 {
  margin: 0;
  text-align: center;
}

hr {
  width: 75%;
}

.timeframes {
  list-style: none;
  margin: 0;
  padding: 0;
  text-align: left;

  .yellow {
    color: yellow;
  }

  .purple {
    color: purple;
  }
}