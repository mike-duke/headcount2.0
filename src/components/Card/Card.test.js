import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card.js';

describe('Card', () => {

  it('should match the snapshot with all appropriate data', () => {
    const stats = {
      2007: 1,
      2008: 0,
      2006: 1,
      2009: 0.5,
      2010: 1
    }
    const wrapper = shallow(<Card location="COLORADO" 
                                  stats={stats}
                                  addDistrictToCompare={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  })

  it('should render a list item with the class of "yellow" if the stat passed in is below 0.5', () => {

  })

  it('should render a list item with the class "purple" if the stat passed in is above 0.5', () => {

  })
})