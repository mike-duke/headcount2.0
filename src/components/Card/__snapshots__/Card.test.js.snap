// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all appropriate data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addDistrictToCompare={[MockFunction]}
    location="COLORADO"
    stats={
      Object {
        "2006": 1,
        "2007": 1,
        "2008": 0,
        "2009": 0.5,
        "2010": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          COLORADO
        </h3>,
        <hr />,
        <ul
          className="timeframes"
        >
          <li>
            2006: 
            <span
              className="purple"
            >
              1
            </span>
          </li>
          <li>
            2007: 
            <span
              className="purple"
            >
              1
            </span>
          </li>
          <li>
            2008: 
            <span
              className="yellow"
            >
              0
            </span>
          </li>
          <li>
            2009: 
            <span
              className="yellow"
            >
              0.5
            </span>
          </li>
          <li>
            2010: 
            <span
              className="purple"
            >
              1
            </span>
          </li>
        </ul>,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              2006: 
              <span
                className="purple"
              >
                1
              </span>
            </li>,
            <li>
              2007: 
              <span
                className="purple"
              >
                1
              </span>
            </li>,
            <li>
              2008: 
              <span
                className="yellow"
              >
                0
              </span>
            </li>,
            <li>
              2009: 
              <span
                className="yellow"
              >
                0.5
              </span>
            </li>,
            <li>
              2010: 
              <span
                className="purple"
              >
                1
              </span>
            </li>,
          ],
          "className": "timeframes",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "2006",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2006: ",
                <span
                  className="purple"
                >
                  1
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "2006: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                  "className": "purple",
                },
                "ref": null,
                "rendered": "1",
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2007",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2007: ",
                <span
                  className="purple"
                >
                  1
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "2007: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                  "className": "purple",
                },
                "ref": null,
                "rendered": "1",
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2008",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2008: ",
                <span
                  className="yellow"
                >
                  0
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "2008: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "0",
                  "className": "yellow",
                },
                "ref": null,
                "rendered": "0",
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2009",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2009: ",
                <span
                  className="yellow"
                >
                  0.5
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "2009: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "0.5",
                  "className": "yellow",
                },
                "ref": null,
                "rendered": "0.5",
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2010",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2010: ",
                <span
                  className="purple"
                >
                  1
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "2010: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                  "className": "purple",
                },
                "ref": null,
                "rendered": "1",
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            COLORADO
          </h3>,
          <hr />,
          <ul
            className="timeframes"
          >
            <li>
              2006: 
              <span
                className="purple"
              >
                1
              </span>
            </li>
            <li>
              2007: 
              <span
                className="purple"
              >
                1
              </span>
            </li>
            <li>
              2008: 
              <span
                className="yellow"
              >
                0
              </span>
            </li>
            <li>
              2009: 
              <span
                className="yellow"
              >
                0.5
              </span>
            </li>
            <li>
              2010: 
              <span
                className="purple"
              >
                1
              </span>
            </li>
          </ul>,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                2006: 
                <span
                  className="purple"
                >
                  1
                </span>
              </li>,
              <li>
                2007: 
                <span
                  className="purple"
                >
                  1
                </span>
              </li>,
              <li>
                2008: 
                <span
                  className="yellow"
                >
                  0
                </span>
              </li>,
              <li>
                2009: 
                <span
                  className="yellow"
                >
                  0.5
                </span>
              </li>,
              <li>
                2010: 
                <span
                  className="purple"
                >
                  1
                </span>
              </li>,
            ],
            "className": "timeframes",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "2006",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2006: ",
                  <span
                    className="purple"
                  >
                    1
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "2006: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1",
                    "className": "purple",
                  },
                  "ref": null,
                  "rendered": "1",
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2007",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2007: ",
                  <span
                    className="purple"
                  >
                    1
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "2007: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1",
                    "className": "purple",
                  },
                  "ref": null,
                  "rendered": "1",
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2008",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2008: ",
                  <span
                    className="yellow"
                  >
                    0
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "2008: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "0",
                    "className": "yellow",
                  },
                  "ref": null,
                  "rendered": "0",
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2009",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2009: ",
                  <span
                    className="yellow"
                  >
                    0.5
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "2009: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "0.5",
                    "className": "yellow",
                  },
                  "ref": null,
                  "rendered": "0.5",
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2010",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2010: ",
                  <span
                    className="purple"
                  >
                    1
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "2010: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1",
                    "className": "purple",
                  },
                  "ref": null,
                  "rendered": "1",
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
